src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts

```diff
@@ -1535,6 +1540,33 @@ export class BadgeClassEditFormComponent extends BaseAuthenticatedRoutableCompon
 		return null;
 	}
 
+	// noEmptyCompetencies(): { emptyCompetencies: Boolean } | null {
+	// 	if(!this.badgeClassForm) return null;
+	// 	const competencies = this.badgeClassForm.controls.competencies.value
+	// 	if (this.checkDuplicateCompetency()) return { emptyCompetencies: true };
+	// }
+
+	// checkDuplicateCompetency(): String | null {
+	// 	if (!this.badgeClassForm) return null;
+
+	// 	const inHandCompetencies = this.badgeClassForm.controls.competencies.value
+	// 		// Hand competencies get added automatically at submitting
+	// 		//filter(c => c.added).
+	// 		.map((c) => c.name);
+
+	// 	const newSelectedAICompetencies = this.badgeClassForm.controls.aiCompetencies.value
+	// 		.map((c, i) => (c.selected ? this.aiCompetenciesSuggestions[i].preferred_label : ''))
+	// 		.filter(String);
+
+	// 	const all = inHandCompetencies.concat(newSelectedAICompetencies);
+	// 	const check = new Set();
+
+	// 	for (const name of all)
+	// 		if (check.has(name)) return name;
+	// 		else check.add(name);
+	// 	return null;
+	// }
+
```

Comment:
Looks like its left over and unused?
Shouldn't make its way to production then :)